File: 백준/Silver/10845. 큐/큐.swift
enum CommandType: String {
    case push
    case pop
    case size
    case empty
    case front
    case back
}

let n = Int(readLine()!)!
var queue: [String] = []
for _ in 0..<n {
    let input = readLine()!.split(separator: " ").map { String($0) }
    guard let command = CommandType(rawValue: input[0]) else {
        break
    }
    
    switch command {
    case .push:
        queue.append(input[1])
    case .pop:
        print(queue.isEmpty ? -1 : queue.removeFirst())
    case .size:
        print(queue.count)
    case .empty:
        print(queue.isEmpty ? 1 : 0)
    case .front:
        print(queue.first ?? -1)
    case .back:
        print(queue.last ?? -1)
    }
}

File: 백준/Silver/10845. 큐/큐.py
import sys

input = sys.stdin.readline

n = int(input())
queue = []

for _ in range(n):
    cmd = input().split()
    
    if cmd[0] == 'push':
        value = int(cmd[1])
        queue.append(value)
    elif cmd[0] == 'pop':
        if queue:
            print(queue.pop(0))
        else:
            print(-1)
    elif cmd[0] == 'size':
        print(len(queue))
    elif cmd[0] == 'empty':
        if queue:
            print(0)
        else:
            print(1)
    elif cmd[0] == 'front':
        if queue:
            print(queue[0])
        else:
            print(-1)
    elif cmd[0] == 'back':
        if queue:
            print(queue[-1])
        else:
            print(-1)